test/lib/openai-mock-server.js

```diff
@@ -69,8 +70,43 @@ function handler(req, res) {
       res.setHeader(key, value)
     }
     res.statusCode = code
-    res.write(JSON.stringify(body))
-    res.end()
+
+    if (payload.stream === true) {
+      // OpenAI streamed responses are double newline delimited lines that
+      // are prefixed with the string `data: `. The end of the stream is
+      // terminated with a `done: [DONE]` string.
+      const parts = body.split(' ')
```

Comment:
splitting the body causes 1 string, which is fine but you have to add that back in above when you push the chunk

Reply:
I changed it locally to this and it works as expected

```js
          if (i < parts.length) {
            const content = (parts.length - 1) === i ? parts[i] : `${parts[i]} `
            const chunk = JSON.stringify({
              id: 'chatcmpl-8MzOfSMbLxEy70lYAolSwdCzfguQZ',
              object: 'chat.completion.chunk',
              // 2023-11-20T09:00:00-05:00
              created: 1700488800,
              model: 'gpt-4',
              choices: [
                {
                  index: 0,
                  finish_reason: null,
                  delta: { role: 'assistant', content }

```